independent-projects/tools/platform-descriptor-resolver-json/src/main/java/io/quarkus/platform/descriptor/resolver/json/QuarkusJsonPlatformDescriptorResolver.java

```diff
@@ -0,0 +1,270 @@
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.graph.Dependency;
+import org.eclipse.aether.resolution.ArtifactDescriptorResult;
+import org.eclipse.aether.resolution.VersionRangeResult;
+import org.eclipse.aether.version.Version;
+
+import com.eclipsesource.json.Json;
+import com.eclipsesource.json.JsonObject;
+import com.eclipsesource.json.JsonValue;
+import com.eclipsesource.json.WriterConfig;
+
+import io.quarkus.bootstrap.resolver.AppModelResolverException;
+import io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver;
+import io.quarkus.platform.descriptor.QuarkusPlatformDescriptor;
+import io.quarkus.platform.descriptor.loader.json.QuarkusJsonPlatformDescriptorLoader;
+import io.quarkus.platform.descriptor.loader.json.QuarkusJsonPlatformDescriptorLoaderContext;
+import io.quarkus.platform.tools.DefaultMessageWriter;
+import io.quarkus.platform.tools.MessageWriter;
+import io.quarkus.platform.tools.ToolsConstants;
+
+/**
+ * Helps resolve the specific or the latest available version of a JSON platform descriptor.
+ */
+public class QuarkusJsonPlatformDescriptorResolver {
+
+    public static final String PROP_PLATFORM_JSON_GROUP_ID = "quarkus.platform.json.groupId";
+    public static final String PROP_PLATFORM_JSON_ARTIFACT_ID = "quarkus.platform.json.artifactId";
+    public static final String PROP_PLATFORM_JSON_VERSION = "quarkus.platform.json.version";
+    public static final String PROP_PLATFORM_JSON_VERSION_RANGE = "quarkus.platform.json.version-range";
+
+    public static QuarkusJsonPlatformDescriptorResolver newInstance() {
+        return new QuarkusJsonPlatformDescriptorResolver();
+    }
+
+    private String jsonGroupId;
+    private String jsonArtifactId;
+    private String jsonVersionRange;
+    private String jsonVersion;
+    private MavenArtifactResolver mvn;
+    private MessageWriter log;
+
+    public QuarkusJsonPlatformDescriptorResolver() {
+    }
+
+    public QuarkusJsonPlatformDescriptorResolver setPlatformJsonGroupId(String groupId) {
+        this.jsonGroupId = groupId;
+        return this;
+    }
+
+    public QuarkusJsonPlatformDescriptorResolver setPlatformJsonArtifactId(String artifactId) {
+        this.jsonArtifactId = artifactId;
+        return this;
+    }
+
+    public QuarkusJsonPlatformDescriptorResolver setPlatformJsonVersionRange(String versionRange) {
+        this.jsonVersionRange = versionRange;
+        return this;
+    }
+
+    public QuarkusJsonPlatformDescriptorResolver setPlatformJsonVersion(String version) {
+        this.jsonVersion = version;
+        return this;
+    }
+
+    public QuarkusJsonPlatformDescriptorResolver setMavenArtifactResolver(MavenArtifactResolver mvn) {
+        this.mvn = mvn;
+        return this;
+    }
+
+    public QuarkusJsonPlatformDescriptorResolver setMessageWriter(MessageWriter msgWriter) {
+        this.log = msgWriter;
+        return this;
+    }
+
+    public QuarkusPlatformDescriptor resolve() {
```

Comment:
Maybe you could split that big method in a few sub private ones?

Reply:
I would if those method would be re-usable in other places. Otherwise, it's not that long and I'm pretty sure it's be changing pretty soon anyway.

Reply:
Sorry I am allergic to long methods 😂 I prefer shorter unit testable ones